catalog/ui/catalog-ui-metacard-sharing-impl/src/main/java/org/codice/ddf/catalog/ui/sharing/ShareableMetacardImpl.java

```diff
@@ -203,8 +205,9 @@ public ShareableMetacardImpl setAccessGroups(Set<String> accessGroups) {
 
   private static boolean isSharingCapable(Metacard metacard) {
     return metacard != null
-        && (metacard.getTags().contains(QUERY_TEMPLATE_TAG)
-            || metacard.getTags().contains(WORKSPACE_TAG)
-            || metacard.getTags().contains(ATTRIBUTE_GROUP_TAG));
+            && (metacard.getTags().contains(QUERY_TEMPLATE_TAG)
+                || metacard.getTags().contains(WORKSPACE_TAG)
+                || metacard.getTags().contains(ATTRIBUTE_GROUP_TAG))
+        || metacard.getTags().contains(SHARING_TAG);
```

Comment:
How could an NPE get thrown here?

Reply:
metacard.getTags() could in theory be null

Reply:
Funny that our JavaDoc says it will return null but the default implementation will not return null.  It still could be overwritten to return null.

https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java#L270

Reply:
Actually it is possible for `metacard` to be null here.

As written, this is evaluated as `(metacard != null && (...)) || metacard.getTags().contains(SHARING_TAG)`. So if the first part of the OR fails (which it could if `metacard` is null) then the second part will be evaluated and `metacard` could be null.